library/std/src/io/error/tests.rs

```diff
@@ -67,3 +67,18 @@ fn test_const() {
     assert!(format!("{:?}", E).contains("\"hello\""));
     assert!(format!("{:?}", E).contains("NotFound"));
 }
+
+#[test]
+fn test_error_packing() {
```

Comment:
Does this get full coverage of the Repr kinds we support? I can see how it would cover OS, Simple, and SimpleMessage, but not Custom.

Reply:
That's covered elsewhere in the file to some degree, but I've beefed the tests up and made them fairly explicit.